### DISCUSSION TITLE
Upgrade from 3.5.3 to 4.1.0

### DISCUSSION THREAD

**Discussion by tkcg2k** [netbox-community/netbox#17432] [Help Wanted!]
Hi, what do i need to do to upgrade from ver 3.5.3 to 4.1.0. 
we use redhat 9
thx in advance

**Comment by candlerb** [comment]
First you need to upgrade to v3.7.8, then you can upgrade to v4.1.0

The full details are in the [upgrade instructions](https://docs.netbox.dev/en/stable/installation/upgrading/) and you'll need to be careful to select the correct instructions for how Netbox was installed in the first place (either from a release tarball or from git).

However, the very safest way is to build a completely new VM with v3.7.8, test it, [export the database](https://netboxlabs.com/docs/netbox/en/stable/administration/replicating-netbox/) from your v3.5.3 system, import it into the new VM, then run `./upgrade.sh` and test it again. Finally upgrade the new VM to v4.1.0.  Doing it this way eliminates any risk of breaking your existing system.
